lib/lotus/repository.rb

```diff
@@ -551,6 +551,42 @@ def transaction(options = {})
         end
       end
 
+
+      # Executes the given raw sql on the adapter.
+      #
+      # Please note that it's only supported by some databases, a `NotImplementedError` will be
+      # raised when the adapter does not responds to the `execute` method.
+      #
+      # For advanced scenarios, please check the documentation of each adapter.
+      #
+      # @see Lotus::Model::Adapters::SqlAdapter#execute
+      #
+      # @since
+      #
+      # @example Basic usage with SQL adapter
+      #   require 'lotus/model'
+      #
+      #   class Article
+      #     include Lotus::Entity
+      #     attributes :title, :body
+      #   end
+      #
+      #   class ArticleRepository
+      #     include Lotus::Repository
+      #   end
+      #
+      #   article = Article.new(title: 'Introducing transactions',
+      #     body: 'lorem ipsum')
+      #
+      #   resultset = ArticleRepository.execute("select * from articles")
+      #   resultset.each_hash do |result|
+      #     result # -> { id: 123, title: "Introducing transactions", body: "lorem ipsum"}
+      #   end
+      def execute(sql)
+        raise NotImplementedError unless @adapter.respond_to?(:execute)
```

Comment:
@peterberkenbosch I'd get rid of this defensive line. Can you please add `#execute` to `Adapters::Abstract` and let it to raise this exception?

That class is a parent for the other two adapters that we support (memory and fs), they will inherit this behavior. Also, `Abstract` serves as a parent or official interface to support in order to be compliant with Lotus::Model.